---
bool success
string error
string pathToFile